clang/lib/CIR/Dialect/Transforms/TargetLowering/Targets/LoweringPrepareX86CXXABI.cpp

```diff
@@ -129,223 +129,244 @@ mlir::Value LoweringPrepareX86CXXABI::lowerVAArgX86_64(
                                           builder, datalayout, valist, ty, loc),
                                       ty));
 
-  auto currentBlock = builder.getInsertionBlock();
-
-  // AMD64-ABI 3.5.7p5: Step 2. Compute num_gp to hold the number of
-  // general purpose registers needed to pass type and num_fp to hold
-  // the number of floating point registers needed.
-
-  // AMD64-ABI 3.5.7p5: Step 3. Verify whether arguments fit into
-  // registers. In the case: l->gp_offset > 48 - num_gp * 8 or
-  // l->fp_offset > 304 - num_fp * 16 go to step 7.
-  //
-  // NOTE: 304 is a typo, there are (6 * 8 + 8 * 16) = 176 bytes of
-  // register save space).
-
-  mlir::Value inRegs;
-  mlir::Value gp_offset_p, fp_offset_p;
-  mlir::Value gp_offset, fp_offset;
-
-  if (neededInt) {
-    gp_offset_p = builder.createGetMemberOp(loc, valist, "gp_offset", 0);
-    gp_offset = builder.createLoad(loc, gp_offset_p);
-    inRegs = builder.getUnsignedInt(loc, 48 - neededInt * 8, 32);
-    inRegs = builder.createCompare(loc, cir::CmpOpKind::le, gp_offset, inRegs);
-  }
+  auto scopeOp = builder.create<cir::ScopeOp>(loc, [&](mlir::OpBuilder
```

Comment:
It'd be best if the lambda just saves the insertion point and then you use `setInsertionPointToEnd` to insert over there. This will be a shorter diff and also helps if we ever have to debug this: block only gets linked after the `create` finishes, you can't dump blocks while investigating crashes happening inside the lambda.